internal/api/v1/handlers/instance.go

```diff
@@ -113,14 +113,16 @@ func (h *InstanceHandler) DeleteInstance(c *fiber.Ctx) error {
 		})
 	}
 
+	// TODO: I think if no instances are provided it should just delete all instances for the job.
+	// In order to do this we need the provider info which will be a DB request from DeleteInstance.
 	if len(req.Instances) == 0 {
 		return c.Status(fiber.StatusBadRequest).JSON(fiber.Map{
 			"error": "at least one instance is required",
 		})
 	}
 
 	// Delete instance using the service
-	job, err := h.service.DeleteInstance(c.Context(), req.ID, req.Name, req.ProjectName, req.Instances)
+	job, err := h.service.DeleteInstance(c.Context(), req.ID, req.InstanceName, req.ProjectName, req.Instances)
```

Comment:
I know its out of the scope of this PR but it is confusing when DeleteInstance returns a job object

Reply:
it could simply just return an error if something fails

Reply:
agreed. I think this is out of scope of this PR though. 
Should track it as part of the job/project clean up you are working on